internal/app/app.go

```diff
@@ -637,45 +655,31 @@ func (c *Config) SetVolumes(volumes []scanner.Volume) {
 }
 
 type ProcessConfig struct {
-	Cmd             []string
-	MachineServices []api.MachineService
-	MachineChecks   map[string]api.MachineCheck
+	Cmd      []string
+	Services []api.MachineService
+	Checks   map[string]api.MachineCheck
 }
 
 func (c *Config) GetProcessConfigs(appLaunching bool) (map[string]ProcessConfig, error) {
 	res := make(map[string]ProcessConfig)
-	processCount := 0
-	if c.Processes != nil {
-		processCount = len(c.Processes)
-	}
-	defaultProcessName := ""
-	firstProcessNameOrDefault := ""
-	if processCount == 1 {
-		for procName := range c.Processes {
-			firstProcessNameOrDefault = procName
-			break
-		}
+	processCount := len(c.Processes)
+	if processCount == 0 {
+		c.Processes = map[string]string{"": ""}
```

Comment:
the original was causing a panic when c.Processes was nil, and this version fixes that. Setting c.Processes here does break assumptions elsewhere that empty processes means no processes were set.  i'll try not overwriting c.Processes and see if that fixes stuff...

Might be good to do some more testing before merging this.

Reply:
thanks for this!